sbi/inference/trainers/npse/npse.py

```diff
@@ -505,46 +535,11 @@ def _loss(
         """
         if self._round == 0 or force_first_round_loss:
             # First round loss.
-            loss = self._neural_net.loss(theta, x)
+            loss = self._neural_net.loss(theta, x, times)
         else:
             raise NotImplementedError(
                 "Multi-round NPSE with arbitrary proposals is not implemented"
             )
 
         assert_all_finite(loss, "NPSE loss")
         return calibration_kernel(x) * loss
-
-    def _converged(self, epoch: int, stop_after_epochs: int) -> bool:
-        """Check if training has converged.
-
-        Unlike the `._converged` method in base.py, this method does not reset to the
-        best model. We noticed that this improves performance. Deleting this method
-        will make C2ST tests fail. This is because the loss is very stochastic, so
-        resetting might reset to an underfitted model. Ideally, we would write a
-        custom `._converged()` method which checks whether the loss is still going
-        down **for all t**.
```

Comment:
this is resolved now with the new ema check?

Reply:
The c2st tests pass with the current settings, at the least!